src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```
@@ -15,16 +15,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Analysis level is a single property that can be used to control both the compiler warning waves
          and enable .NET analyzers. Valid values are 'none', 'latest', 'preview', or a version number  -->
 
-    <!-- Automatically set AnalysisLevel to 5 if targeting .net 5 and nothing else specified -->
+    <!-- If not specified and greater than or equal to .NET 5, default the AnalysisLevel to the the MAJ.MIN version -->
     <AnalysisLevel Condition="'$(AnalysisLevel)' == '' And
                               '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
-                              $([MSBuild]::VersionEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">5.0</AnalysisLevel>
-
-    <!-- Automatically set AnalysisLevel if not specified -->
-    <AnalysisLevel Condition="'$(AnalysisLevel)' == '' And
-                              (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
-                               $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '6.0'))) Or
-                               '$(AnalysisMode)' != '')">latest</AnalysisLevel>
+                              $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">$(_TargetFrameworkVersionWithoutV)</AnalysisLevel>
```

```suggestion
                              $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">latest</AnalysisLevel>
```

This will match the documentation in https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#latest-updates, and also fix https://github.com/dotnet/docs/issues/29054 :)